function/templates/TriggerAuthentication.yaml

```
@@ -0,0 +1,9 @@
+{{- if .Values.triggerAuthName }}
+apiVersion: keda.k8s.io/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{- .Values.triggerAuthName }}
```

Any reason this wouldn't work? Might be good to remove in other places where needed as well.
```suggestion
  name: {{ .Values.triggerAuthName }}
```

---

Its worth making it like 
  {{ tpl .Values. triggerAuthName $ }} as apps can configure them as
  triggerAuthName : {{ .Release.Name }}-send-letter. If not, it could clash with other releases especially in preview.